src/Microsoft.AspNet.Routing/RouteValueDictionary.cs

```diff
@@ -289,7 +302,126 @@ public bool TryGetValue(string key, out object value)
                 throw new ArgumentNullException(nameof(key));
             }
 
-            return _dictionary.TryGetValue(key, out value);
+            if (_dictionary != null)
+            {
+                return _dictionary.TryGetValue(key, out value);
+            }
+            else if (_properties != null)
+            {
+                for (var i = 0; i < _properties.Length; i++)
+                {
+                    var property = _properties[i];
+                    if (Comparer.Equals(property.Name, key))
+                    {
+                        value = property.ValueGetter(_value);
+                        return true;
+                    }
+                }
+
+                value = null;
+                return false;
+            }
+            else
+            {
+                value = null;
+                return false;
+            }
+        }
+
+        private void EnsureWritable()
+        {
+            if (_dictionary == null && _properties == null)
+            {
+                _dictionary = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            }
+            else if (_dictionary == null)
+            {
+                _dictionary = new Dictionary<string, object>(_properties.Length + 1, StringComparer.OrdinalIgnoreCase);
```

Comment:
Yeah, I think I'll just drop the +1. I didn't find any examples of a dictionary being resized during testing. 

Dictionary doesn't use your capacity verbatim anyway.